client/src/pages/recording-view/components/annotation/annotation-viewer.tsx

```diff
@@ -2,11 +2,64 @@
 // Copyright (c) 2023 Marc Lichtman
 // Licensed under the MIT License
 
-import React, { Fragment, useCallback, useMemo } from 'react';
-import { Layer, Rect, Text } from 'react-konva';
+import React, { Fragment, useEffect, useCallback, useMemo, useState } from 'react';
+import { Layer, Rect, Text, Group } from 'react-konva';
 import { TILE_SIZE_IN_IQ_SAMPLES } from '@/utils/constants';
 import { Annotation, SigMFMetadata } from '@/utils/sigmfMetadata';
 import { useSpectrogramContext } from '../../hooks/use-spectrogram-context';
+import { Html } from 'react-konva-utils';
+
+interface AnnotationLabelEditorProps {
+  labelText: string;
+  y: number;
+  x: number;
+  id: string;
+  setEditAnnotationLabelY: (y: number) => void;
+  setEditAnnotationLabelText: (text: string) => void;
+  setEditAnnotationLabelChanged: (flag: boolean) => void;
+}
+
+export const AnnotationLabelEditor = ({
+  labelText,
+  y,
+  x,
+  id,
+  setEditAnnotationLabelY,
+  setEditAnnotationLabelText,
+  setEditAnnotationLabelChanged,
+}: AnnotationLabelEditorProps) => {
+  const onKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      setEditAnnotationLabelY(null);
+      if (e.target.value == '') {
+        setEditAnnotationLabelText(labelText);
+      } else {
+        if (labelText !== e.target.value) {
+          setEditAnnotationLabelText(e.target.value);
+          setEditAnnotationLabelChanged(true);
+        }
+      }
+    }
+  };
+  const top = y + 'px';
+  const left = x + 'px';
+  const msgTop = (y - 30).toString() + 'px';
+  return (
+    <Group>
+      <Html>
+        <textarea
+          placeholder={labelText}
+          style={{ top: top, left: left, width: '200px', fontSize: '16px', position: 'absolute' }}
+          rows={1}
+          className={'text-base-100'}
+          id={id}
+          onKeyDown={onKeyDown}
+        ></textarea>
+      </Html>
+      <Text text="Hit Enter to Finish" fontFamily="serif" fontSize={24} x={x} y={y - 30} fill={'black'}></Text>
```

Comment:
Do we need to be using Konva elements here? I would have thought a textarea/input would be fine

Reply:
I tried really hard to use the second textarea for the "press enter to finish", but couldn't figure out how to get the font in the right color. Then I noticed that I don't really need the textarea for that. A konva Text works just fine.